secant/Features/App/App.swift

```diff
@@ -43,6 +65,86 @@ extension AppReducer {
+            
+            /// Checking presense of stored wallet in the keychain and presense of database files in documents directory.
+        case .checkWalletInitialisation:
+            var walletInitialisationState: InitialisationState = .uninitialized
+
+            let fileManager = FileManager()
+            // TODO: use updated dependency from PR #217 (https://github.com/zcash/secant-ios-wallet/pull/217)
+            let keysPresent = environment.recoveryPhraseStorage.areKeysPresent()
+            
+            do {
+                // TODO: Create a dependency to handle database files for the SDK, issue #220 (https://github.com/zcash/secant-ios-wallet/issues/220)
+                // TODO: use database URL from the same issue #220
+                let documentsURL = try FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
+                let dataDatabaseURL = documentsURL.appendingPathComponent("ZcashSDK.defaultDataDbName", isDirectory: false)
+                let attributes = try fileManager.attributesOfItem(atPath: dataDatabaseURL.path)
+                let databaseFilesPresent = attributes.isEmpty
+                
+                switch (keysPresent, databaseFilesPresent) {
+                case (false, false):
+                    walletInitialisationState = .uninitialized
+                case (false, true):
+                    walletInitialisationState = .keysMissingFilesPresent
+                case (true, false):
+                    walletInitialisationState = .keysPresentFilesMissing
+                case (true, true):
+                    walletInitialisationState = .initialized
+                }
+            } catch CocoaError.fileNoSuchFile, CocoaError.fileReadNoSuchFile {
+                walletInitialisationState = keysPresent ? .keysPresentFilesMissing : .uninitialized
+            } catch {
+                walletInitialisationState = .failure(ErrorWrapper(error: error))
+            }
+            
+            return Effect(value: .walletInitialisationResult(walletInitialisationState))
+
+            /// All possible responses to the `.checkWalletInitialisation` action result.
+            /// `uninitialized` results in the start of the Onboarding flow for the user.
+        case .walletInitialisationResult(let walletInitialisationState):
+            switch walletInitialisationState {
+            case .uninitialized:
+                return Effect(value: .updateRoute(.onboarding))
+                    .delay(for: 3, scheduler: environment.mainQueue)
+                    .eraseToEffect()
+            case .initialized, .keysPresentFilesMissing:
+                return Effect(value: .initialiseApp)
+            case .keysMissingFilesPresent:
+                state.appInitialisationState = walletInitialisationState
+                // TODO: error we need to handle, issue #221 (https://github.com/zcash/secant-ios-wallet/issues/221)
```

Comment:
[non-blocking] 

This is currently an error but this state could be a "feature" when we support "Viewing Key Only" wallets.

Reply:
Noted, seems like there's no action point at the moment but might be in the future.